magicblock-accounts/tests/commit_delegated.rs

```diff
@@ -86,13 +86,15 @@ async fn test_commit_two_delegated_accounts_one_needs_commit() {
     generate_validator_authority_if_needed();
     let commit_needed_pubkey = Pubkey::new_unique();
     let commit_needed_account = generate_account(&commit_needed_pubkey);
-    let commit_needed_account_shared =
+    let mut commit_needed_account_shared =
         AccountSharedData::from(commit_needed_account.clone());
+    commit_needed_account_shared.set_delegated(true);
```

Comment:
Ditto, let's not change this behavior unless we have a reason why it needs to be done.
I want to avoid changing more than necessary in order to just keep the current cloning version working until we have the new one.